src/plots/linePlot.ts

```diff
@@ -348,6 +348,44 @@ namespace Plottable.Plots {
     }
 
     /**
+     * Gets the Entities that intersect the Bounds.
+     *
+     * @param {Bounds} bounds
+     * @returns {PlotEntity[]}
+     */
+    public entitiesIn(bounds: Bounds): PlotEntity[];
+    /**
+     * Gets the Entities that intersect the area defined by the ranges.
+     *
+     * @param {Range} xRange
+     * @param {Range} yRange
+     * @returns {PlotEntity[]}
+     */
+    public entitiesIn(xRange: Range, yRange: Range): PlotEntity[];
+    public entitiesIn(xRangeOrBounds: Range | Bounds, yRange?: Range): PlotEntity[] {
+      let dataXRange: Range;
+      let dataYRange: Range;
+      if (yRange == null) {
+        let bounds = (<Bounds> xRangeOrBounds);
+        dataXRange = { min: bounds.topLeft.x, max: bounds.bottomRight.x };
+        dataYRange = { min: bounds.topLeft.y, max: bounds.bottomRight.y };
+      } else {
+        dataXRange = (<Range> xRangeOrBounds);
+        dataYRange = yRange;
+      }
+
+      const xProjector = Plot._scaledAccessor(this.x());
+      const yProjector = Plot._scaledAccessor(this.y());
+      return this.entities().filter((entity) => {
```

Comment:
perf note. I noticed that plot's `.entities` method does a couple linear time passes on the data. Notably, it uses `_lightweightEntities` then maps them to "heavyweight" entities, which as far as i can tell, just adds a d3 selection. For perf, we should maybe consider using lightweight entities when filtering, then convert to `PlotEntities` as a post process. Lots of other methods use `this.entities()` so, probably not worth in this PR, but for the future.